app/assets/javascripts/discourse/app/components/composer-messages.js

```diff
@@ -47,6 +51,8 @@ export default class ComposerMessages extends Component {
     this.appEvents.off("composer:find-similar", this, this._findSimilar);
     this.appEvents.off("composer-messages:close", this, this._closeTop);
     this.appEvents.off("composer-messages:create", this, this._create);
+
+    clearRecipientNamesCache();
```

Comment:
Is the cache actually tied to this specific instance of the component? This change seems to imply that (since it's cleared on teardown). If so, could it just be an instance variable?

If the idea to share the cache _across_ different instances of the same component, maybe the cache can be tied to the owner instead?

Reply:
O good point, this can just be a property of the component since the cache is mostly a perf optimisation to ensure we're not hitting the server with the same request if the recipient names stays the same. I don't think it has to be shared between different instances of the composer. I'll follow up with another PR.